middleware.ts

```diff
@@ -65,6 +65,13 @@ export async function middleware(req: NextRequest) {
     data: { session },
   } = await supabase.auth.getSession();
 
+  if (session?.user && req.nextUrl.pathname === "/workspaces") {
```

Comment:
There are several conditions in the middleware, four to be exact that start with `if (session?.user &&`. Not for this PR, but it'd make sense to consolidate so it's one branch for that condition and handle the subconditions within it.